src/main/scala/nl/knaw/dans/easy/license/Command.scala

```diff
@@ -15,16 +15,67 @@
  */
 package nl.knaw.dans.easy.license
 
-import org.slf4j.LoggerFactory
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream, FileOutputStream, OutputStream}
+
+import com.yourmediashelf.fedora.client.FedoraClient
 import nl.knaw.dans.easy.license.{CommandLineOptions => cmd}
+import org.slf4j.LoggerFactory
+import rx.lang.scala.Observable
+import rx.schedulers.Schedulers
+
+import scala.language.postfixOps
 
 object Command {
   val log = LoggerFactory.getLogger(getClass)
 
   def main(args: Array[String]): Unit = {
     log.debug("Starting command line interface")
-    val ps = cmd.parse(args)
+    implicit val parameters = cmd.parse(args)
+
+    new FileOutputStream(parameters.outputFile)
+      .usedIn(run)
+      .doOnTerminate {
+        // close LDAP at the end of the main
+        log.debug("closing ldap")
+        parameters.ldap.close()
+      }
+      .subscribe(_ => {}, e => log.error("An error was caught in main:", e), () => log.debug("completed"))
+
+    Schedulers.shutdown()
+  }
+
+  /**
+    * Create a license agreement and write it the `outputStream`. The `parameters` object a.o. contain
+    * the datasetID and the depositorID (optional).
+    *
+    * ``Notice:`` neither the `outputStream` nor the `LdapContext` in `parameters` are closed at the
+    * end of this operation.
```

Comment:
It is not the responsibility of this function to do so. I give the `outputStream` and the `LdapContext` as parameters to this function, so the one who calls it is responsible for closing these resources. The notice in this ScalaDoc was added as a reminder for the user of this function to not forget to close these resources.